### PR TITLE
Move GPS injection message handling up to GCS_MAVLINK

### PR DESCRIPTION
**Pull request by peterbarker**
Based on this other PR: https://github.com/ArduPilot/ardupilot/pull/6667

### CODE REVIEW

File: libraries/AP_GPS/AP_GPS.cpp [hunk 1]
```diff
@@ -747,6 +756,10 @@ void AP_GPS::handle_msg(const mavlink_message_t *msg)
         handle_gps_rtcm_data(msg);
         return;
     }
+    if (msg->msgid == MAVLINK_MSG_ID_GPS_INJECT_DATA) {
```
Comment (OXINARF) [inline libraries/AP_GPS/AP_GPS.cpp]: Nitpicking: else if

File: libraries/AP_GPS/AP_GPS.h [hunk 2]
```diff
@@ -346,9 +346,7 @@ class AP_GPS
     // lock out a GPS port, allowing another application to use the port
     void lock_port(uint8_t instance, bool locked);
 
-    //Inject a packet of raw binary to a GPS
-    void inject_data(uint8_t *data, uint16_t len);
-    void inject_data(uint8_t instance, uint8_t *data, uint16_t len);
+    void handle_gps_inject(const mavlink_message_t *msg);
```
Comment (OXINARF) [inline libraries/AP_GPS/AP_GPS.h]: Why is this one public?

File: libraries/AP_GPS/AP_GPS.cpp (hunk 1 above)
Comment (peterbarker) [inline libraries/AP_GPS/AP_GPS.cpp]: Heh.  Here we go.

Changing it to a switch to avoid argument  :-)

File: libraries/AP_GPS/AP_GPS.h (hunk 2 above)
Comment (peterbarker) [inline libraries/AP_GPS/AP_GPS.h]: Sheer stupidity can't be ruled out.  Fixed.

**Comment by peterbarker** [comment]
@OXINARF Fixes pushed up.  Including a change for your objections to the perfectly cromulent "ibject"

**Comment by OXINARF** [comment]
Merged, thanks!

**Pull request merged**